packages/monorepo-builder/src/FileSystem/ComposerJsonProvider.php

```diff
@@ -28,6 +28,11 @@ final class ComposerJsonProvider
      */
     private $composerJsonFactory;
 
+    /**
+     * @var SmartFileInfo[]|null;
+     */
+    private ?array $packageComposerFiles = null;
```

Comment:
Nullable array should be avoided, as they break on foreach. Empty array would do

Reply:
Ok done